dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/graph/WorkflowGraphTopologyLogicalVisitor.java

```diff
@@ -143,7 +143,11 @@ private void doVisitationInSubGraph(Set<String> subGraphNodes) {
             if (!visitedTaskCodes.contains(taskName)) {
                 visitedTaskCodes.add(taskName); // Record the nodes
                 final Set<String> successors = workflowGraph.getSuccessors(taskName);
-                if (subGraphNodes.contains(taskName)) {
+                // In TASK_ONLY mode, pass empty successors since there are no subsequent nodes
+                if (taskDependType == TaskDependType.TASK_ONLY) {
+                    final Set<String> emptySuccessors = new HashSet<>();
+                    visitFunction.accept(taskName, emptySuccessors);
+                } else {
```

Comment:
The `subGraphNodes` should be only one node in this case, so only node in `subGraphNodes` will be visit, is this bug caused by here?